src/handlers/shared/start.ts

```diff
@@ -129,16 +116,34 @@ export async function start(context: Context, issue: Context["payload"]["issue"]
   return { output: "Task assigned successfully" };
 }
 
+async function fetchUserId(context: Context, username: string) {
+  try {
+    const user = await context.octokit.rest.users.getByUsername({ username });
+    return user.data.id;
+  } catch (e) {
+    return null;
+  }
+}
+
 async function handleTaskLimitChecks(username: string, context: Context, maxConcurrentTasks: number, logger: Context["logger"], sender: string) {
   const openedPullRequests = await getAvailableOpenedPullRequests(context, username);
   const assignedIssues = await getAssignedIssues(context, username);
 
   // check for max and enforce max
-  if (assignedIssues.length - openedPullRequests.length >= maxConcurrentTasks) {
-    throw logger.error(username === sender ? "You have reached your max task limit" : `${username} has reached their max task limit`, {
+
+  if (Math.abs(assignedIssues.length - openedPullRequests.length) >= maxConcurrentTasks) {
```

Comment:
won't this mean if someone has 5 PRs but 2 issues assigned, they can't start a new task? I guess this is open for debate if they should be allowed to start a new task

Reply:
The original implementation allowed for productive contributors to work on two tasks concurrently:
- If any reviewer approved the pull, +1 
- If reviewers take longer than 24 hours, +1 
- If any reviewer requested changes, -1

Reply:
2 (assigned) - 5 (PRs) = 3 === < 5 (maxTasks), is my understanding of that line of code. So someone with 0 assignedIssues and 5 PRs cannot be assigned anymore tasks.

> I'll make another task for this

To refactor this whole feature, so this requested change can be resolved for now?

Reply:
> We should also do +1 task if the PR author made necessary changes requested by the reviewer but are waiting for reviewer's next review and it's been more than 24 hours.
> 
> 
> 
> Currently the code fetches reviews but that endpoint returns all reviews even multiple reviews from the same person, so we need use only the last review by the same person but also check requested reviews. If a reviewer has requested changes but is being requested for review more than 24 hours, it should be considered +1 task.
> 
> 
> 
> I'll make another task for this

This is tricky. People generally aren't very discipline about changing it to a draft when they are continuing to edit it. What signal will be used to indicate that they are finished? We also can't always rely on them to formally request a new review from the UI. Most contributors will just tag the reviewers to check again. 

I don't think this is going to be viable.